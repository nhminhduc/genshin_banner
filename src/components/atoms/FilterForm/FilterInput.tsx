type FilterInputProps = {
  className?: string;
  id?: string;
  innerRef?: React.LegacyRef<HTMLInputElement>;
  name: string;
  onChange(event: React.ChangeEvent<HTMLInputElement>): void;
  type?: string;
  value: string;
};

const FilterInput = ({
  className,
  id,
  innerRef,
  name,
  onChange,
  type = "text",
  value,
}: FilterInputProps) => (
  <div>
    <input
      className={className}
      id={id}
      name={name}
      onChange={onChange}
      ref={innerRef}
      type={type}
      value={value}
    />
  </div>
);

export default FilterInput;
