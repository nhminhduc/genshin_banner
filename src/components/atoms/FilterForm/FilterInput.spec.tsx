import { render } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";

import FilterInput from "./FilterInput";

describe("FilterInput test", () => {
  test("FilterInput load", () => {
    const mockFunction = vi.fn();
    const { container } = render(
      <FilterInput
        name="FilterInput test"
        onChange={mockFunction}
        value="FilterInput test"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
