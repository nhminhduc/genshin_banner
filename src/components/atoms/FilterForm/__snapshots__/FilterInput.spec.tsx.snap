// Vitest Snapshot v1

exports[`FilterInput test > FilterInput load 1`] = `
<div>
  <div>
    <input
      class="p-2"
      defaultvalue="FilterInput test"
      name="FilterInput test"
      type="text"
    />
  </div>
</div>
`;
